src/Abstract/DynamicLink.jl

```diff
@@ -156,6 +158,7 @@ function find_library(rpath::RPath, soname::AbstractString)
     return soname
 end
 
+@mustimplement isdynamic(oh::ObjectHandle)
```

Comment:
Put this in `ObjectHandle.jl`, like right [here](https://github.com/staticfloat/ObjectFile.jl/blob/98d7b327448456df024ab87243520f972c715e16/src/Abstract/ObjectHandle.jl#L216).  Add a docstring to the abstract definition, similar to `islibrary()`, add it to the list of methods [here](https://github.com/staticfloat/ObjectFile.jl/blob/98d7b327448456df024ab87243520f972c715e16/src/Abstract/ObjectHandle.jl#L46-L56).